cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/SamplingUtils.java

```diff
@@ -346,124 +315,70 @@ private static boolean skipBuildingBrokerMetricSample(BrokerLoad brokerLoad, int
   }
 
   /**
-   * Build a wrapper around the topic with the given desired properties and {@link #DEFAULT_CLEANUP_POLICY}.
+   * Create a Kafka consumer for retrieving reported Cruise Control metrics.
+   * The consumer uses {@link String} for keys and {@link CruiseControlMetric} for values.
    *
-   * @param topic The name of the topic.
-   * @param partitionCount Desired partition count.
-   * @param replicationFactor Desired replication factor.
-   * @param retentionMs Desired retention in milliseconds.
-   * @return A wrapper around the topic with the given desired properties.
+   * @param configs The configurations for Cruise Control.
+   * @return A new Kafka consumer
    */
-  public static NewTopic wrapTopic(String topic, int partitionCount, short replicationFactor, long retentionMs) {
-    if (partitionCount <= 0 || replicationFactor <= 0 || retentionMs <= 0) {
-      throw new IllegalArgumentException(String.format("Partition count (%d), replication factor (%d), and retention ms (%d)"
-                                                       + " must be positive for the topic (%s).", partitionCount,
-                                                       replicationFactor, retentionMs, topic));
+  public static Consumer<String, CruiseControlMetric> createMetricConsumer(Map<String, ?> configs) {
+    // Get bootstrap servers
+    String bootstrapServers = (String) configs.get(METRIC_REPORTER_SAMPLER_BOOTSTRAP_SERVERS);
+    if (bootstrapServers == null) {
+      bootstrapServers = bootstrapServers(configs);
     }
-
-    NewTopic newTopic = new NewTopic(topic, partitionCount, replicationFactor);
-    Map<String, String> config = new HashMap<>(2);
-    config.put(RetentionMsProp(), Long.toString(retentionMs));
-    config.put(CleanupPolicyProp(), DEFAULT_CLEANUP_POLICY);
-    newTopic.configs(config);
-
-    return newTopic;
-  }
-
-  /**
-   * Add config altering operations to the given configs to alter for configs that differ between current and desired.
-   *
-   * @param configsToAlter A set of config altering operations to be populated.
-   * @param desiredConfig Desired config value by name.
-   * @param currentConfig Current config.
-   */
-  private static void maybeUpdateConfig(Set<AlterConfigOp> configsToAlter, Map<String, String> desiredConfig, Config currentConfig) {
-    for (Map.Entry<String, String> entry : desiredConfig.entrySet()) {
-      String configName = entry.getKey();
-      String targetConfigValue = entry.getValue();
-      ConfigEntry currentConfigEntry = currentConfig.get(configName);
-      if (currentConfigEntry == null || !currentConfigEntry.value().equals(targetConfigValue)) {
-        configsToAlter.add(new AlterConfigOp(new ConfigEntry(configName, targetConfigValue), AlterConfigOp.OpType.SET));
-      }
+    // Get group id
+    long randomToken = RANDOM.nextLong();
```

Comment:
See comment about group id

Reply:
Addressed this similar to https://github.com/linkedin/cruise-control/pull/1332#discussion_r496353768.